tmt/steps/provision/__init__.py

```diff
@@ -74,6 +74,15 @@
     from tmt._compat.typing import TypeAlias
 
 
+#: How many seconds to wait for a connection to succeed after guest boot.
+#: This is the default value tmt would use unless told otherwise.
+DEFAULT_CONNECT_TIMEOUT = 2 * 60
+
+#: How many seconds to wait for a connection to succeed after guest boot.
+#: This is the effective value, combining the default and optional envvar,
+#: ``TMT_CONNECT_TIMEOUT``.
+CONNECT_TIMEOUT: int = configure_constant(DEFAULT_CONNECT_TIMEOUT, 'TMT_CONNECT_TIMEOUT')
```

Comment:
I would recommend adding a function similar to `default_reconnect_waiting` and `default_reboot_waiting`, e.g. `default_connect_waiting`, and then use it in the `assert_reachable` function.

I'm not 100 % sure how our usage of timeouts, ticks and waitings will look like, I'm trying to follow this approach for now.